frontend/src/actions/index.js

```diff
@@ -1,45 +1,103 @@
-import api from "../api";
+import api, { non_auth_api } from '../api';
 import {
-    FETCH_LESSON,
-    FETCH_QUESTIONS,
-    FETCH_CHOICES,
-    CREATE_ANSWER,
-    FETCH_ANSWERS,
-    FETCH_CATEGORIES
-} from "./types";
+	REGISTER,
+	LOGIN,
+	LOGOUT,
+	LOGIN_ERROR,
+	REGISTER_ERROR,
+	LOGOUT_ERROR,
+	FETCH_LESSON,
+	FETCH_QUESTIONS,
+	FETCH_CHOICES,
+	CREATE_ANSWER,
+	FETCH_ANSWERS,
+	FETCH_CATEGORIES,
+} from './types';
+import history from '../history';
+
+export const register = (data) => async (dispatch) => {
+	try {
+		const response = await non_auth_api.post(`api/register/`, data);
+
+		dispatch({ type: REGISTER, payload: response.data });
+		history.push('/login');
+	} catch (e) {
+		let error = {
+			username:
+				e.response.data?.username?.length > 0
```

Comment:
Apply to all related
```suggestion
				!!e.response.data?.username?.length
```